### ISSUE TITLE
Too many redirects with 18.0.1.3 (Umleitungsfehler)

### ISSUE THREAD

**Issue by diehummel** [nextcloud/server#19747]
### Steps to reproduce
1. just try yourself on https://drive.hummel-web.at
2.
3.

### Expected behaviour
Login should be visible

### Actual behaviour
Error (Umleitungsfehler in german). Think this means too many redirects in englisch.

### Server configuration

**Operating system:**
linux (webhoster, not my own server)

**Web server:**
apache
**Database:**
mysql 
**PHP version:**
PHP Version 7.3.14

**Nextcloud version:** (see Nextcloud admin page)
18.0.1.3

**Updated from an older Nextcloud/ownCloud or fresh install:**
updated from 15
Also tried with a fresh install. same situation.

**Where did you install Nextcloud from:**

**Signing status:**
<details>
<summary>Signing status</summary>

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```
</details>

**List of activated apps:**
Enabled:                                                                                        
  - accessibility: 1.4.0                                                                        
  - bruteforcesettings: 1.5.0                                                                   
  - cloud_federation_api: 1.1.0                                                                 
  - dav: 1.14.0                                                                                 
  - federatedfilesharing: 1.8.0                                                                 
  - federation: 1.8.0                                                                           
  - files: 1.13.1                                                                               
  - files_external: 1.9.0                                                                       
  - files_pdfviewer: 1.7.0                                                                      
  - files_rightclick: 0.15.2                                                                    
  - files_sharing: 1.10.1                                                                       
  - files_trashbin: 1.8.0                                                                       
  - files_versions: 1.11.0                                                                      
  - files_videoplayer: 1.7.0                                                                    
  - logreader: 2.3.0                                                                            
  - lookup_server_connector: 1.6.0                                                              
  - nextcloud_announcements: 1.7.0                                                              
  - oauth2: 1.6.0                                                                               
  - password_policy: 1.8.0
  - photos: 1.0.0                                                                               
  - privacy: 1.2.0                                                                              
  - provisioning_api: 1.8.0                                                                     
  - serverinfo: 1.8.0                                                                           
  - settings: 1.0.0                                                                             
  - sharebymail: 1.8.0                                                                          
  - support: 1.1.0                                                                              
  - survey_client: 1.6.0                                                                        
  - text: 2.0.0                                                                                 
  - theming: 1.9.0                                                                              
  - twofactor_backupcodes: 1.7.0                                                                
  - updatenotification: 1.8.0                                                                   
  - viewer: 1.2.0                                                                               
  - workflowengine: 2.0.0                                                                       
Disabled:                                                                                       
  - activity                                                                                    
  - admin_audit                                                                                 
  - comments                                                                                    
  - encryption                                                                                  
  - firstrunwizard                                                                              
  - notifications                                                                               
  - recommendations                                                                             
  - systemtags                                                                                  
  - user_ldap
**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

or 

Insert your config.php content here. 
Make sure to remove all sensitive content such as passwords. (e.g. database password, passwordsalt, secret, smtp password, …)
```
<?php
$CONFIG = array (
  'instanceid' => 'ociuqt50bfmo',
  'passwordsalt' => 'q5817BMuoFQRYi/e3sw34m67jhPdfu',
  'secret' => 'tvt9Whyw2OkVGeccmg1Z/1BwxG5/MikzQXRqtxoXkx4sEC1V',
  'trusted_domains' => 
  array (
    0 => 'drive.hummel-web.at',
  ),
  'datadirectory' => '/home/.sites/874/site/web/subdomains/drive/data/',
  'overwritehost' => 'drive.hummel-web.at',
  'overwriteprotocol' => 'https',
  'dbtype' => 'mysql',
  'version' => '18.0.1.3',
  'dbname' => '',
  'dbhost' => '.world4you.com',
  'dbtableprefix' => 'oc_',
  'dbuser' => '',
  'dbpassword' => '',
  'logtimezone' => 'Europe/Berlin',
  'installed' => true,
  'theme' => '',
  'tempdirectory' => '/home/.sites/874/site/tmp',
  'integrity.check.disabled' => false,
  'singleuser' => false,
  'updatechecker' => true,
  'knowledgebaseenabled' => true,
  'session_lifetime' => 86400,
  'versions_retention_obligation' => 'auto',
  'updater.server.url' => 'https://updates.nextcloud.com/updater_server/',
  'updater.release.channel' => 'stable',
  'updater.secret' => '$',
  'appstoreenabled' => true,
  'log_level' => 2,
  'logfile' => '/home/.sites/874/site/web/subdomains/drive/data/nextcloud_debug.log',
  'maintenance' => false,
);

<summary>LDAP config</summary>

```
With access to your command line run e.g.:
sudo -u www-data php occ ldap:show-config
from within your Nextcloud installation folder

Without access to your command line download the data/owncloud.db to your local
computer or access your SQL server remotely and run the select query:
SELECT * FROM `oc_appconfig` WHERE `appid` = 'user_ldap';


Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```
</details>

### Client configuration
**Browser:**
Firefox (latest)
**Operating system:**
Windows 10

#### Web server error log
<details>
<summary>Web server error log</summary>

```
There is no log entry in the apache log and no in the nextcloud log.
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
There is no log entry in the apache log and no in the nextcloud log.
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```
</details>

**Comment by kesselb** [comment]
Thank you for taking the time to report a problem :+1: 

As this seems to be a setup issue I would like to ask you to raise your question at https://help.nextcloud.com

**Comment by Kaliumhexacyanoferrat** [comment]
After upgrading the `nextcloud-fpm` image today, I stumbled over the same issue. I looked into the HTTP communication, which yields the following results:

1. The first request will redirect to the login page which is normal, and properly set the cookies for the PHP session
2. The second request to `/login` will include the newly set cookie values. The nextcloud container responds with 302 (not my reverse proxy).

If you request `/login` as your first request without having cookies set, the page will properly render. As the 302 redirect is generated by the nextcloud container, I suspect this to be a bug instead of a missconfiguration as well.
